backend/indexer/internal/indexer/rpc.go

```diff
@@ -171,9 +160,9 @@ func (i *Indexer) getTransactionsReceipts(hash []string) ([]*types.TransactionRe
 		params = append(params, []interface{}{hash[i]}, []interface{}{hash[i]})
 	}
 
-	var rpcResponses []client.RPCResponse
-	if err := i.client.CallRpcBatched(methods, params, &rpcResponses); err != nil {
-		return nil, fmt.Errorf("failed to execute batch RPC call: %w", err)
+	rpcResponses, err := i.retryRPCBatched(methods, params)
+	if err != nil {
+		return nil, fmt.Errorf("%w", err)
```

Comment:
nit: not much of a point wrapping this error without a message, either return it as is, or add some context on the failure in the error message, e.g.

```go
return nil, fmt.Errorf("failed to get transaction (%s) receipts: %w", hash, err)